src/Microsoft.VisualStudio.LanguageServices.Razor/Editor/DefaultTextBufferFactoryService.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor.Editor
+{
+    [System.Composition.Shared]
+    [Export(typeof(TextBufferFactoryService))]
+    internal class DefaultTextBufferFactoryService : TextBufferFactoryService
+    {
+        private static readonly object RazorTextBufferInitializationKey = new object();
+
+        private readonly VisualStudioDocumentTrackerFactory _documentTrackerFactory;
+        private readonly VisualStudioRazorParserFactory _parserFactory;
+        private readonly BraceSmartIndenterFactory _braceSmartIndenterFactory;
+
+        [ImportingConstructor]
+        public DefaultTextBufferFactoryService(
+            VisualStudioDocumentTrackerFactory documentTrackerFactory,
```

Comment:
There are now factories for each of the initialization pieces (tracker, parser, smartindenter). I felt that it was much cleaner to isolate the dependencies of factories and ultimately keep this class straightforward.

Reply:
ok